### PR TITLE
Chart cell  - geodata

### PR DESCRIPTION
**Pull request by cristineguadelupe**
Screen Shot 2023-01-23 at 11 26 51

### CODE REVIEW

File: lib/assets/chart_cell/main.js [hunk 1]
```diff
@@ -281,11 +378,19 @@ export function init(ctx, payload) {
...
+        return this.layers.length > 1 && !onlyGeo;
       },
       unavailableDataVariable() {
         return !this.dataVariables.includes(this.layers[0].data_variable);
       },
+      hasGeoLayer() {
+        const geoLayer = this.layers.find(
+          (layer) => layer.chart_type === "geoshape"
+        );
+        return geoLayer ? true : false;
```
Comment (jonatanklosko) [inline lib/assets/chart_cell/main.js:392]:
```suggestion
        return this.layers.some((layer) => layer.chart_type === "geoshape");
```

File: lib/assets/chart_cell/main.js [hunk 2]
```diff
@@ -281,11 +378,19 @@ export function init(ctx, payload) {
         return !this.layers[0].data_variable;
       },
       hasLayers() {
-        return this.layers.length > 1;
+        const onlyGeo =
+          this.layers.length === 2 && this.layers[0].chart_type === "geoshape";
```
Comment (jonatanklosko) [inline lib/assets/chart_cell/main.js:382]:
```suggestion
        const onlyGeo = this.layers.length === 2 && this.isGeoLayer(0);
```

File: lib/assets/chart_cell/main.js [hunk 3]
```diff
@@ -76,14 +76,52 @@ export function init(ctx, payload) {
...
+                    fieldClass="field"
+                  />
                 </div>
-                <div class="row">
-                  <div class="input-icon-container">
+                <div class="row" v-if="layer.chart_type !== 'geoshape'">
+                  <BaseSelect
+                    v-if="layer.geodata"
+                    name="latitude_field"
+                    label="latitude"
```
Comment (josevalim) [inline lib/assets/chart_cell/main.js:118]:
```suggestion
                    label="Latitude"
```

File: lib/assets/chart_cell/main.js [hunk 4]
```diff
@@ -134,7 +172,16 @@ export function init(ctx, payload) {
                       </template>
                     <FieldSettings/>
                   </div>
-                  <div class="input-icon-container">
+                  <BaseSelect
+                    v-if="layer.geodata"
+                    name="longitude_field"
+                    label="longitude"
```
Comment (josevalim) [inline lib/assets/chart_cell/main.js:178]:
```suggestion
                    label="Longitude"
```

**Comment by josevalim** [comment]
* Layer is missing on the second layer
* Add new chart types instead of a checkbox

**Review by josevalim — APPROVED** [review]

**Pull request merged**